src/main/java/filter/Parser.java

```diff
@@ -126,6 +127,8 @@ private FilterExpression parseKeyword(Token token) {
 
     private int getInfixPrecedence(Token token) {
         switch (token.getType()) {
+        case SEMICOLON:
+            return Precedence.SHORTER_DISJUNCTION;
```

Comment:
I think a better solution would be to just pass everything to `parseSeparatedContent` (and still retain the special case to sort qualifier) and let the semantic error handler in the future to handle it.